plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+"""Parser for Google Drive Sync log files."""
+
+from __future__ import unicode_literals
+
+import logging
+
+import pyparsing
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+
+class GoogleDriveSyncLogEventData(events.EventData):
+  """Google Drive Sync log event data.
+
+  Attributes:
+    log_level (str): logging level of event such as "DEBUG", "WARN", "INFO",
+        "ERROR".
+    message (str): log message.
+    pid (int): process ID of process which logged event.
+    source_code (str): filename:line_number of source file which logged event.
+    thread (str): colon-separated thread identifier in the form "ID:name"
+        which logged event.
+    time (str): date and time of the log entry event with timezone offset.
+  """
+
+  DATA_TYPE = 'gdrive_sync:log:line'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(GoogleDriveSyncLogEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.time = None
+    self.log_level = None
+    self.pid = None
+    self.thread = None
+    self.source_code = None
+    self.message = None
+
+
+class GoogleDriveSyncLogParser(text_parser.PyparsingMultiLineTextParser):
+  """Parses events from Google Drive Sync log files."""
+
+  NAME = 'gdrive_synclog'
+
+  DESCRIPTION = 'Parser for Google Drive Sync log files.'
+
+  _ENCODING = 'utf-8'
+
+  # Increase the buffer size, as log messages are often many lines of python
```

Comment:
Explain why you override the default buffer size.

Reply:
Expanded on the comment here, LMK if you think it needs more context.